Interview process (#5)
Ada-Developers-Academy/core-interlude: interview-process/interview-introduction.md

![don't panic](/assets/interview-process_interview-introduction_dont-panic.jpg)

The Ada curriculum and community provide a strong support network to help you succeed in the interview process.

## Goals

- Describe the most common steps in the application process.
- Understand hiring terminology such as:
  - Whiteboard Interview
  - Coding Challenge
  - Phone Screening

## Vocabulary and Synonyms

| Vocab	|  Definition	| Synonym | How to Use in a Sentence |
|---	|---	|---	|---	|
|  Whiteboard Interview  	|  A problem solving exercise where a job applicant is asked to solve a programming or design problem on a whiteboard. 	|    Technical Interview	|   I have a whiteboard interview next week, so I need a buddy to practice with. 	|
| Phone Screen | An initial telephone or internet chat with a job candidate to ensure the candidate's suitability for an in-peron interview.  This could involve a hiring manager, but is more likely to involve a recruiter. | &nbsp; | I have a phone screening with a recruiter at Dynamic Widgets in 3 hours. | 

> | Coding Challenge | A "homework" assignment given to job candidates to assess their programming skills and practices such as regularly committing and writing tests. |  | I'm trying to write tests for my Dynamic Widgets coding challenge assignment. 

Reviewer: ```suggestion
| Coding Challenge | A "homework" assignment given to job candidates to assess their programming skills and practices, such as regularly committing and writing tests. |  | I'm trying to write tests for my Dynamic Widgets coding challenge assignment. 
```

## The Typical Tech Hiring Process

When applying for a junior developer position in the United States, applicants typically follow these steps.

1. Applying for a position
1. Phone or video chat screening
1. In-person or video interview
1. Offer & Negotiation
1. Acceptance

> These steps are typical in the process, but the process can vary greatly.  Companies may not engage in the phone screening or might require additional interviews, or provide homework in the form of a coding challenge.

Reviewer: ```suggestion
These steps are typical in the process, but the process can vary greatly.  Companies may not engage in phone screening, or they may require additional interviews. They may provide homework in the form of a coding challenge as well.
```

### Application

> A junior developer can apply for a position in many ways, this can include an online application form on the company website or a job-listing site like [Indeed](https://www.indeed.com/q-Apply-jobs.html).  Sometimes it can also include talking to a recruiter, a colleague at the company, or a hiring manager at a local meet-up.  Universities like the University of Washington hold job-fairs.  There are a lot of ways to get in the door as a junior developer, but the most effective strategies typically involve networking.  

Reviewer: ```suggestion
A junior developer can apply for a position in many ways. This can include an online application form on the company website or a job-listing site, like [Indeed](https://www.indeed.com/q-Apply-jobs.html).  Sometimes it can also include talking to a recruiter, a colleague at the company, or a hiring manager at a local meet-up.  Universities like the University of Washington hold job-fairs.  

There are a lot of ways to get in the door as a junior developer, but the most effective strategies typically involve networking.  
```

> When possible, candidates should take advantage of opportunities to meet with engineers, managers and recruiters either in person or electronically.  Online applications turned into hiring systems are much easier to ignore than a human being with a connection to the team or company.

Reviewer: ```suggestion
When possible, candidates should take advantage of opportunities to meet with engineers, managers, and recruiters, either in person or electronically.  Online applications turned into hiring systems are much easier to ignore than a human being with a connection to the team or company.
```

### Phone Screen

> A phone screen, despite the name might be either on the phone or via an online platform such as Skype or Google Hangout.  Candidates will likely initially talk to a recruiter, but the call may involve a hiring manager.  Candidates may or may not be asked technical questions, but the goal of a phone screen is generally to narrow down the list of candidates for on site interviews.  Generally the discussion centers on work experience, projects worked on and how the candidate's interests align with the company and team applied for.

Reviewer: ```suggestion
Despite the name, a phone screen can be either on the phone or via an online platform, such as Skype or Google Hangout.  Candidates will likely talk to a recruiter initially, but the call may involve a hiring manager.  

Candidates may or may not be asked technical questions, but the goal of a phone screen is to narrow down the list of candidates for onsite interviews.  Generally, the discussion centers on work experience, projects worked on, and how the candidate's interests align with the company and team applied for.
```

> Before a phone screen it is a good idea to:

Reviewer: ```suggestion
Before a phone screen, it is a good idea to:
```

- Review the job listing and take notes
- Review the resume as the interviewer will likely ask about work experience

> - Prepare some questions to help determine how attractive the position

Reviewer: ```suggestion
- Prepare some questions to help determine how attractive the position is
```

- Find a quiet place with good connectivity.

### In-person Interview

> A technical in-person interview can vary greatly in length and process.  Typically however a candidate will have some initial time to greet and introduce yourself to the interviewer(s) with some brief chitchat.  

Reviewer: ```suggestion
A technical in-person interview can vary greatly in length and process; however, typically a candidate will have some initial time to greet and introduce yourself to the interviewer(s) with some brief chitchat.  
```

> Then the interviewer will begin the technical portion of the interview.  This is often a _whiteboard interview_, meant to test the candidate's ability to solve new problems and communicate their thoughts.  Candidates are traditionally given a problem and a whiteboard and asked to develop a solution and explain it to the interviewer.

Reviewer: ```suggestion
Then the interviewer will begin the technical portion of the interview.  This is often a _whiteboard interview_, meant to test the candidate's ability to solve new problems and communicate their thoughts.  Candidates are traditionally given a problem and a whiteboard, asked to develop a solution, and then explain it to the interviewer.
```

> In recent times, candidates are often interviewed in an online technical interview using Zoom, Meet, Cisco WebEx and a shared online code editor such as [Feplit](https://replit.com/), [Coderpad](https://coderpad.io/), [HackerRank](https://hackerrank.com/) or [VS Code LiveShare](https://docs.microsoft.com/en-us/visualstudio/liveshare/use/vscode).  In a process similar to an in-person interview candidates are presented with a programming problem and work through a solution live with the interviewers demonstrating problem solving, and communication skills.

Reviewer: ```suggestion
In recent times, candidates are often interviewed online, using Zoom, Meet, Cisco WebEx, and use a shared online code editor, such as [Feplit](https://replit.com/), [Coderpad](https://coderpad.io/), [HackerRank](https://hackerrank.com/), or [VS Code LiveShare](https://docs.microsoft.com/en-us/visualstudio/liveshare/use/vscode).  In a process similar to an in-person interview, candidates are presented with a programming problem and are asked to work through a solution live with the interviewers, demonstrating problem solving and communication skills.
```

> <!-- available callout types: info, success, warning, danger, secondary  -->

Reviewer: ```suggestion
```

### !callout-success

## Mock Interviews

> As an Ada student you will be matched up for a practice interview, called a "mock interview" with an experienced hiring manager or engineer.  They will give you a sample interview problem and conduct a practice interview with you.  After the "mock interview" the interviewer will write up a set of feedback for you to help you prepare for your actual internship interviews.

Reviewer: ```suggestion
As an Ada student, you will be matched up for a practice interview called a "mock interview" with an experienced hiring manager or engineer.  They will give you a sample interview problem and conduct a practice interview with you.  After the "mock interview," the interviewer will write up feedback for you to help you prepare for your actual internship interviews.
```

Remember:  

![don't panic](/assets/interview-process_interview-introduction_dont-panic.jpg)

The mock interview is designed to give you practice and experience with a technical interview and provide you useful feedback.

### !end-callout

> <!-- available callout types: info, success, warning, danger, secondary  -->

Reviewer: ```suggestion
```
